### DISCUSSION TITLE
Does my show-only-matching-columns command make sense?

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1826] [Q&A]
Hey, I recently implemented a command that I call `show-only-matching-columns`. It's working well for me, but I wonder whether I'm missing something, i.e. maybe there's an existing command that does something similar.

When I type `gc`, I get a box to type a regex. Then VisiData shows only the columns that match that regex, hiding the rest of the columns. This excludes key columns, which are always kept visible. This is useful for me because I have ~100 columns and I want to see only a handful at a time.

Code:

```
@Sheet.api
def show_only_matching_columns(sheet: Sheet, regex_string: str) -> None:
    regex_flags = sum(getattr(re, f.upper()) for f in sheet.options.regex_flags) 
    pattern = re.compile(regex_string, regex_flags)
    columns_to_show = []
    columns_to_hide = []
    for column in sheet.columns:
        if pattern.search(column.name) or (column in sheet.keyCols):
            columns_to_show.append(column)
        else:
            columns_to_hide.append(column)

    for column_to_hide in columns_to_hide:
        column_to_hide.hide()
    vd.unhide_cols(columns_to_show, sheet.visibleRows)


Sheet.addCommand(
    'gc', # Todo: When Insert is supported, change to 'Ins-c'.
    'show-only-matching-columns',
    'show_only_matching_columns(input("select regex: ", type="regex", defaultLast=True))',
    'Show only the columns matching the given regex, hiding other columns.'
)
```

**Comment by saulpw** [comment]
Yes, this makes sense to me!  Nothing jumps out at me as incorrect.  and especially if it works for you, that's the thing that really matters.  Thanks for sharing!

**Reply by cool-RR** [reply]
I'm happy to hear that. Now that I think about it, it would be cool if I had "and" and "or" modes for this command, or a separate command for each mode. e.g. I already limited the columns to a certain set, and now I want to show columns that match a new regex in addition to the ones already shown. I'll think how to organize the key bindings for that...

**Reply by frosencrantz** [reply]
Whatever happened with https://github.com/saulpw/visidata/issues/1702?  Was there ever a resolution?  It seems related to this discussion.  I would like some good ways to wrangle many columns.
